core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -239,13 +239,35 @@ private Path unzip(Path zip, Path pluginsDir) throws IOException {
         return target;
     }
 
+    /** Checks that the plugin contain a top-level folder titled `elasticsearch` and only one sub-folder titled after the name of the plugin. */
+    private Path findPluginRoot(Path root) throws Exception {
```

Comment:
This method is not necessary. With the code as is, we would be extracting the entire zip, but only using the one directory from it. Instead, we should do checks when extracting, see the `unzip` method. There we can have a prefix check like:

```
if (entry.getName().startsWith("elasticsearch/") == false) {
    // only extract the elasticsearch directory
    continue;
}
Path targetFile = target.resolve(entry.getName().substring("elasticsearch/".size()));
```

This will unzip everything in the `elasticsearch` directory directly into the temp installation directory, and all the other plugin cli installation code can work as-is.